datumaro/util/test_utils.py

```diff
@@ -155,6 +208,9 @@ def test_save_and_load(test, source_dataset, converter, test_dir, importer,
     if target_dataset is None:
         target_dataset = source_dataset
 
-    if not compare:
+    if not compare and kwargs.get("dimension") is Dimensions.dim_3d:
+        compare = compare_datasets_3d
+        del kwargs["dimension"]
+    elif not compare:
```

Comment:
It seems like the sole purpose of the `dimension` argument is to select a different default comparison function, which raises the question: why even add that argument? It's not any more difficult for the client of the function to pass `compare=compare_datasets_3d` than it is to pass `dimension=Dimensions.dim_2d`.

Reply:
The only benefit I see is the higher level of abstraction when the enum is used.